handlers/dialog360/dialog360.go

```diff
@@ -283,8 +283,7 @@ func (h *handler) processCloudWhatsAppPayload(ctx context.Context, channel couri
 				}
 
 				// create our message
-				ev := h.Backend().NewIncomingMsg(channel, urn, text, clog).WithReceivedOn(date).WithExternalID(msg.ID).WithContactName(contactNames[msg.From])
-				event := h.Backend().CheckExternalIDSeen(ev)
```

Comment:
Added that in 57c1ea15654341fd08be489ae7a8f37222a06689

Reply:
This was to mainly dedupe by external ID, we do that in the loop since whatsapp can send multiple message in one request

Reply:
In case on message is sent multiple times in different requests, that was not concerned by duplicates in the same request, but having that is probably nicer too